### PR TITLE
fix ducc scores and add tests

### PR DESCRIPTION
**Pull request by aboft**

### CODE REVIEW

File: backend/database/DuccScores.ts [hunk 1]
```diff
@@ -8,7 +8,8 @@ class DatabaseDuccUtils {
       // remove any zero width spaces or control chars from gonzobot
       scoreString = scoreString.replace(/[\x02|\u200B]/g, '').trim();
       const score: string[] = scoreString.split(':');
-      const duccs: number = parseInt(score[1].trim());
+      const duccs: number = parseInt(score[1].trim().replace(/[^0-9]/g, ''));
+      console.log(duccs);
```
Comment (LoLei) [inline backend/database/DuccScores.ts]: Debug log

File: backend/tests/DatabaseDuccScores.test.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  await ircMessageProcessor.processIrcMessage(mockFriendScores);
+  await ircMessageProcessor.processIrcMessage(mockKillerScores);
+});
+
+test('insert or update ducc scores', async () => {
+  const friendScore = ['aboft: 1234', 'asmodean: 1,324,23'];
+  const killerScore = ['audron: 5,443', 'alexendoo: 44'];
+  await DatabaseDuccUtils.insertOrUpdateDuccScores(friendScore, 'friend');
+  await DatabaseDuccUtils.insertOrUpdateDuccScores(killerScore, 'killer');
+});
```
Comment (LoLei) [inline backend/tests/DatabaseDuccScores.test.ts]: These tests should have assertions, like [here](https://github.com/r-lmr/lmrdashboard/blob/lmrdashboard/backend/tests/DataBaseUserUtils.test.ts#L22). I see the last test case in that file doesn't have one either though.

Either return value or calledWith assertions I guess.

**Review by LoLei — CHANGES_REQUESTED** [review]

File: backend/irc/IrcMessageProcessor.ts [hunk 3]
```diff
@@ -160,7 +160,7 @@ class IrcMessageProcessor {
   private async processJoin(ircMessage: IrcMessage): Promise<void> {
     const nick = ircMessage.prefix && ircMessage.prefix.split('!')[0];
     if (nick != this.joinConfig.user) {
-      console.log(nick, nick!.slice(1), nick![0] === '@' || '%' || '+');
+      //console.log(nick, nick!.slice(1), nick![0] === '@' || '%' || '+');
```
Comment (LoLei) [inline backend/irc/IrcMessageProcessor.ts]: By removing the debug line I didn't mean commenting it out :P

**Review by LoLei — CHANGES_REQUESTED** [review]

File: backend/irc/IrcMessageProcessor.ts (hunk 3 above)
Comment (LoLei) [reply backend/irc/IrcMessageProcessor.ts]: That's a different one as well apparently

Comment (aboft) [reply backend/irc/IrcMessageProcessor.ts]: this debug log has been in there for a while and I just caught it. Maybe we would use it in the future (hopeful but improbable). It is removed.

File: backend/database/DuccScores.ts (hunk 1 above)
Comment (aboft) [reply backend/database/DuccScores.ts]: Banished!

File: backend/tests/DatabaseDuccScores.test.ts (hunk 2 above)
Comment (aboft) [reply backend/tests/DatabaseDuccScores.test.ts]: we will make another PR to address the test and use this PR to fix the bug.

**Review by LoLei — APPROVED** [review]

**Pull request merged**